### ISSUE TITLE
Configuring pythontex for ST3 editor

### ISSUE THREAD

**Issue by KSokol79** [gpoore/pythontex#188]
Hi all,

I am having trouble configuring the package for use in my thesis .tex files. I am using Sublime Text 3, TeXLive distro and LaTeXTools to compile my documents. I have double checked that pythontex is supplied within TeXLive, installed Python 3.9.6 and Pygments. Also made sure to use both caption and pythontex packages in my document preamble.

However when I am building my document running programs as follows "pdflatex -> bibtex -> pythontex -> pdflatex -> pdflatex", the log file shows pythontex is being executed and .pytxmcr file is created, but I have no output in my file.

I've only tested with the simplest message using \begin{pycode} environment. The document is using a custom document class (essentially a modified report class).

Would somebody be able to help diagnose where the problem lies?
@gpoore

**Comment by gpoore** [comment]
Have you tried compiling on the command line, outside ST3? That could be a good way to determine if this is a TeX Live or an ST3 configuration issue.

**Comment by KSokol79** [comment]
@gpoore I've been recommended the same thing on stackexchange, currently trying to work out what exactly to do, only rarely do I do anything in command prompt, so don't really know step by step. I assume I have to navigate to the folder containing the .tex file and then run the same sequence?

**Comment by gpoore** [comment]
Yes, you want to open the command prompt, navigate to the folder (use `cd` to change directory), and then run all the programs one at a time (`pdflatex` command, `bibtex` command, and so on). For the commands, you will need to use the name of the .tex file...the ST3 build configuration may be using some sort of placeholder instead.

**Comment by KSokol79** [comment]
I have ran `pdflatex Thesis.tex` in my directory and it completed fine producing a PDF accordingly (still without `pythontex` output as you'd expect), but with `pythontex Thesis.tex` I am getting the following error:

``C:\texlive\2021\bin\win32\runscript.tlu:935:` C:\texlive\2021\bin\win32\runscript.tlu:902: program not found (not part of TeX Live): python.exe``
which I assume is a problem with the Python installation? Should I reinstall? I wonder why ST3 wouldn't flag this up...

**Comment by gpoore** [comment]
The error message is saying that Python is not installed on the system or cannot be found.

The `python` package in TeX Live is separate software that is not related to this.

Are you sure you have Python installed? You can try running `python --version` on the command prompt to find out. It is also possible that Python is installed, but not in PATH, in which case you may need to edit your environment variables or otherwise modify configuration.

**Comment by KSokol79** [comment]
100% it is. 
But the `python --version` command indeed confirmed something's not right.
I've just removed it and will try reinstalling in PATH.

Before I do that, can I check this does not break Anaconda distribution? 
I am currently using some of the apps within it (Jupyter, Spyder) and would rather not have to figure out how to get them to work outside of it.

**Comment by gpoore** [comment]
When you install Anaconda, there is an option to add it to your PATH. If Anaconda is the only Python you are installing under Windows, then adding it to PATH typically shouldn't cause issues. You can add Python to PATH by editing the environment variables; reinstalling isn't necessary though it is an option.

**Comment by KSokol79** [comment]
So I reinstalled Python 3.9.6 making sure to add it to PATH and the command prompt version check is now working properly.
I've also checked that `Pygments` is where it ought to be.

The problem persists - no output from `pycode` environment in the compiled PDF.
What to check next?

**Comment by gpoore** [comment]
It looks like ST3 is using its own PATH. You might take a look at https://stackoverflow.com/questions/37017701/where-is-path-in-sublime-text-3-for-windows?rq=1, and may want to find some other resources on customizing ST3 to use the user or system PATH.

**Comment by KSokol79** [comment]
I had a brief look but I must say I don't follow. I don't understand the relationships between different PATHs and dont want to break the existing installations. I'll keep reading, but would appreciate any further advice. Otherwise I'll probably try and use a different editor or command line when compiling `pythontex`, or resort to importing pre-made figures & tables.

**Comment by gpoore** [comment]
I have no experience with Sublime Text, so I can't help configure that.

You might try restarting the computer, just in case Sublime Text is still using the old version of PATH before you reinstalled Python. Otherwise, if everything works at the command prompt, then you'll have to try to determine what ST3 is using for its PATH and how to modify that.
